tests/testthat/test-preprocess_species.R

```
@@ -13,7 +13,22 @@ test_that("preprocess_species runs on test dataset", {
   expect_no_error(a <- preprocess_species("example_data", hdf5 = FALSE))
   expect_no_error(validate_BirdFlow(a, allow_incomplete = TRUE))
   expect_error(validate_BirdFlow(a))
-  expect_true(all((ext(a)[, ] %% xres(a)) < 1e-9))  # Test if origin is at 0, 0
```

This test is making sure the cells align with the origin and I think is important to keep.   It has nothing to do with your code.

I have two thoughts:
1.  I could add code to preprocess species to check if the extent and cell size numbers are close to integers and round them if they are. It wouldn't help when the cell size is floating point (say some sort of spherical coordinates) but would help with all practical applications we've had to date.
2.  We could lower the tolerance on the check to allow the test to pass as is.  I can't see the failed test output anymore but if I recall the difference was very small but not quite as small as this test requires.

---

I think I will leave the current modification as it is -- To suppress the error. I'm not deleting the test -- just using another logic to validate.

---

Sorry I missed the added lines!

Yours is definitely better.

Your test passes if you are within the tolerance either above or below the exact number, whereas mine only passes if you are above but within the tolerance.  
 What's weird is when it was failing before it was for a value that was slightly too big given the tolerance so, I think, should have failed either test.   Definitely leave yours though, it's both theoretically and empirically working better. 

Thanks!